core/market/src/rest_api.rs

```diff
@@ -9,15 +9,18 @@ use serde::Deserialize;
 
 use ya_client::model::ErrorMessage;
 
-use crate::db::model::{AgreementId, OwnerType, ProposalId, ProposalIdParseError, SubscriptionId};
+use crate::db::model::{
+    AgreementId, AppSessionId, OwnerType, ProposalId, ProposalIdParseError, SubscriptionId,
+};
+use chrono::{DateTime, Utc};
 
 pub(crate) mod common;
 mod error;
 pub(crate) mod provider;
 pub(crate) mod requestor;
 
-const DEFAULT_EVENT_TIMEOUT: f32 = 0.0; // seconds
-const DEFAULT_QUERY_TIMEOUT: f32 = 12.0;
+const DEFAULT_EVENT_TIMEOUT: f32 = 5.0; // seconds
+const DEFAULT_QUERY_TIMEOUT: f32 = 5.0;
```

Comment:
below you can find examlpe how to handle invalid user input for `appSessionId` ie. len > 100 chars in this case